src/components/encoding-pane/filter-shelf.tsx

```diff
@@ -58,31 +68,128 @@ class FilterShelfBase extends React.Component<FilterShelfProps, {}> {
     });
   }
 
+  protected filterModifyTimeUnit(e: any, index: number, domain: number[]) {
+    const {handleAction} = this.props;
+    let timeUnit = e.target.value;
+    if (timeUnit === '-') {
+      timeUnit = undefined;
+    }
+
+    handleAction({
+      type: FILTER_MODIFY_TIME_UNIT,
+      payload: {
+        index,
+        timeUnit
+      }
+    });
+    if (!timeUnit) {
```

Comment:
I'm wondering if some of these logic down here should be handled as a part of `FILTER_MODIFY_TIME_UNIT`.  

Doing so would at least make it easier to write unit tests, but the trade off is that filterReducer will probably need to take schema as input just like we do for shelf.
(I think that's fine.  Otherwise, we should similarly get rid of `schema` from `shelfReducer` too.) 

I'm not sure though.  If you disagree we can discuss